### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	<-ctx.Done()
+	// Background context was canceled. Do whatever cleanup necessary.
+}
+
+func handleDeployment(req router.Request, resp router.Response) error {
+	// Act on the deployment, which can be retrieved via req.Object
+	// If you need to create new objects based on the deployment, use resp.Objects()
+	// A req.Client and req.Ctx are provided for instances when you need to directly interact with the Kubernetes API.
+	return nil
+}
```
Comment (tylerslaton) [inline README.md:41]: Could we add a slightly more detailed example demonstrating the entire I/O flow here?

**Review by tylerslaton — COMMENTED** [review]
Super needed PR, thank you for writing this up.

What would your thoughts be on including a section showing off `Middlewares`?

File: README.md (hunk 1 above)
Comment (thedadams) [reply README.md]: Added a more detailed example.

**Comment by thedadams** [comment]
Can we add something related to middleware in a separate PR?

**Comment by tylerslaton** [comment]
Sure, not a blocker for me. I would be happy to add it even.

**Pull request merged**